app/models/patron_request.rb

```diff
@@ -55,6 +55,11 @@ def barcode=(barcode)
     self.barcodes = [barcode]
   end
 
+  def holdable_recallable_items
+    patron = Folio::Patron.find_by(patron_key: patron_id)
+    items_in_location.filter { |item| item.recallable?(patron) && item.holdable?(patron) }
+  end
+
   private
```

Comment:
What do you think about extracting `patron` and memoize the result of this calculation (I'm assuming calculating the circ rules for each item could get expensive..) ? 

```suggestion
  def holdable_recallable_items
    @holdable_recallable_items ||= items_in_location.filter { |item| item.recallable?(patron) && item.holdable?(patron) }
  end
  
  def patron
      @patron ||= Folio::Patron.find_by(patron_key: patron_id)
  end

  private
```

Reply:
I was thinking the same thing Friday about patron. I think we are going to be needing it for when we submit so I think its a good idea.